net/release-notes/release-notes-2021/aspose-slides-for-net-21-1-release-notes/_index.md

```diff
@@ -0,0 +1,197 @@
+|SLIDESNET-40686|High memory consumption while loading and saving pptx|Bug|
+|SLIDESNET-40073|Text spacing differs from PPTX to PDF|Bug|
+
+## **Public API Changes**
+
+### **Support of the shape points editing has been added** ###
+
+Support of the shape points editing has been added. New classes, interfaces, enums, and **[GeometryShape](https://apireference.aspose.com/slides/net/aspose.slides/geometryshape)** methods have been added.
+
+**New classes:**
+- [PathSegment](https://apireference.aspose.com/slides/net/aspose.slides/pathsegment)
+- [GeometryPath](https://apireference.aspose.com/slides/net/aspose.slides/geometrypath)
+- [ShapeUtil](https://apireference.aspose.com/slides/net/aspose.slides.util/shapeutil)
+
+**New interfaces:**
+- [IPathSegment](https://apireference.aspose.com/slides/net/aspose.slides/ipathsegment)
+- [IGeometryPath](https://apireference.aspose.com/slides/net/aspose.slides/igeometrypath)
+
+**New enums:**
+- [PathFillModeType](https://apireference.aspose.com/slides/net/aspose.slides/pathfillmodetype)
+- [PathCommandType](https://apireference.aspose.com/slides/net/aspose.slides/pathcommandtype)
+
+**Public methods have been added to the GeometryShape class and its descendants:**  
+- IGeometryPath[] [GeometryShape.GetGeometryPaths](https://apireference.aspose.com/slides/net/aspose.slides/geometryshape/methods/getgeometrypaths)()
+- void [IGeometryShape.SetGeometryPath](https://apireference.aspose.com/slides/net/aspose.slides/igeometryshape/methods/setgeometrypath)
+- void [IGeometryShape.SetGeometryPaths](https://apireference.aspose.com/slides/net/aspose.slides/igeometryshape/methods/setgeometrypaths)
+
+Feature description:
+
+Customization of the shape geometry assumes editing points of an existing shape. 
+
+![Edit shape points](1_editPoints_PP.png)
+
+To provide the mentioned functionality [GeometryPath](https://apireference.aspose.com/slides/net/aspose.slides/geometrypath) class and [IGeometryPath](https://apireference.aspose.com/slides/net/aspose.slides/igeometrypath) interface have been added. [GeometryPath](https://apireference.aspose.com/slides/net/aspose.slides/geometrypath) instance represents a geometry path of the [IGeometryShape](https://apireference.aspose.com/slides/net/aspose.slides/igeometryshape) object. 
+
+To retrieve [GeometryPath](https://apireference.aspose.com/slides/net/aspose.slides/geometrypath) from the [IGeometryShape](https://apireference.aspose.com/slides/net/aspose.slides/igeometryshape) instance [GeometryPath](https://apireference.aspose.com/slides/net/aspose.slides/geometrypath) method has been added. Please note that shapes may be built from a few smaller shapes (e.g. an "equal" sign) so this method returns an array of [IGeometryPath](https://apireference.aspose.com/slides/net/aspose.slides/igeometrypath) objects. 
+
+To set [GeometryPath](https://apireference.aspose.com/slides/net/aspose.slides/geometrypath) to the shape has been added two methods: 

```

Comment:
@yury-knml 
The standard word order in English is: Subject + Verb + Object. So, please, follow such order: X method has been added, Y class has been implemented, two methods have been added.
Please, fix it in all the article.